baseplate/secrets/fetcher.py

```diff
@@ -282,13 +282,19 @@ def is_about_to_expire(self):
     def get_secret(self, secret_name):
         """Get the value and expiration time of a named secret."""
         logger.debug("Fetching secret %r.", secret_name)
-        response = self.session.get(
-            urljoin(self.base_url, posixpath.join("v1", secret_name)),
-            headers={
-                "X-Vault-Token": self.token,
-            },
-        )
-        response.raise_for_status()
+        try:
+            response = self.session.get(
+                urljoin(self.base_url, posixpath.join("v1", secret_name)),
+                headers={
+                    "X-Vault-Token": self.token,
+                },
+            )
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            logger.error("Vault client failed to get secret: %s, %s",
```

Comment:
nit: I think a : between the format strings would be easier to read:

    Vault client failed to get secret: secrets/foo/bar, Not found

vs.

    Vault client failed to get secret: secret/foo/bar: Not found